indexing-service/src/main/java/io/druid/indexing/common/task/IndexTask.java

```diff
@@ -151,6 +165,8 @@ public IndexTask(
     super(
         // _not_ the version, just something uniqueish
         makeId(id, ingestionSchema),
+        makeId(id, ingestionSchema),
```

Comment:
These three makeId's are not guaranteed to fire in the same millisecond. You'll need to orchestrate it such that makeId fires once, and that value is re-used.